src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -0,0 +1,2952 @@
+    member_flag_consistency(plain_map.key_flags());
+    map_key_type_identifier_consistency(*plain_map.key_identifier());
+}
+
+void TypeObjectUtils::type_identifier_consistency(
+        const TypeIdentifier& type_identifier)
+{
+    switch (type_identifier._d())
+    {
+        case TK_NONE:
+            throw InvalidArgumentError("Inconsistent TypeIdentifier: non-initialized");
+
+        case TI_STRING8_SMALL:
+        case TI_STRING16_SMALL:
+            string_sdefn_consistency(type_identifier.string_sdefn());
+            break;
+
+        case TI_STRING8_LARGE:
+        case TI_STRING16_LARGE:
+            string_ldefn_consistency(type_identifier.string_ldefn());
+            break;
+
+        case TI_PLAIN_SEQUENCE_SMALL:
+            seq_sdefn_consistency(type_identifier.seq_sdefn());
+            break;
+
+        case TI_PLAIN_SEQUENCE_LARGE:
+            seq_ldefn_consistency(type_identifier.seq_ldefn());
+            break;
+
+        case TI_PLAIN_ARRAY_SMALL:
+            array_sdefn_consistency(type_identifier.array_sdefn());
+            break;
+
+        case TI_PLAIN_ARRAY_LARGE:
+            array_ldefn_consistency(type_identifier.array_ldefn());
+            break;
+
+        case TI_PLAIN_MAP_SMALL:
+            map_sdefn_consistency(type_identifier.map_sdefn());
+            break;
+
+        case TI_PLAIN_MAP_LARGE:
+            map_ldefn_consistency(type_identifier.map_ldefn());
+            break;
+
+        case TI_STRONGLY_CONNECTED_COMPONENT:
+            // No inconsistency rule apply.
+            EPROSIMA_LOG_ERROR(XTYPES_TYPE_REPRESENTATION, "StronglyConnectedComponents not yet supported.");
+            break;
+
+        // Primitive TypeIdentifiers/ExtendedTypeDefn/EquivalenceHash: no inconsistency rule apply.
+        case EK_COMPLETE:
+        case EK_MINIMAL:
+        default:
+            break;
+    }
+}
+
+void TypeObjectUtils::applied_annotation_parameter_seq_consistency(
+        const AppliedAnnotationParameterSeq& applied_annotation_parameter_seq)
+{
+    std::set<NameHash> param_hashes;
+    for (size_t i = 0; i < applied_annotation_parameter_seq.size(); i++)
+    {
+        param_hashes.insert(applied_annotation_parameter_seq[i].paramname_hash());
+        if (param_hashes.size() != (i + 1))
+        {
+            throw InvalidArgumentError("Repeated annotation parameters in the sequence");
+        }
+    }
+}
+
+void TypeObjectUtils::applied_annotation_type_identifier_consistency(
+        const TypeIdentifier& annotation_type_id)
+{
+    TypeObjectPair type_objects;
+    ReturnCode_t ret_code = DomainParticipantFactory::get_instance()->type_object_registry().get_type_object(
+        annotation_type_id, type_objects);
+    if (ReturnCode_t::RETCODE_OK == ret_code)
+    {
+        if (type_objects.complete_type_object._d() != TK_ANNOTATION)
+        {
+            throw InvalidArgumentError("AppliedAnnotation TypeIdentifier does not correspond with an Annotation type");
```

Comment:
```suggestion
            throw InvalidArgumentError("Annotation TypeIdentifier does not correspond with an Annotation type");
```
Just so the error msg is the same.

Reply:
I have added the `Applied` to the other messages. I think is clearer.

Reply:
Done in c0c629be4